pkg/kubelet/volumemanager/cache/desired_state_of_world.go

```diff
@@ -226,13 +232,22 @@ func (dsw *desiredStateOfWorld) AddPodToVolume(
 	}
 
 	if _, volumeExists := dsw.volumesToMount[volumeName]; !volumeExists {
+		var sizeLimit *resource.Quantity
+		if volumeSpec.Volume != nil {
+			if util.IsLocalEphemeralVolume(*volumeSpec.Volume) {
+				_, podLimits := apiv1resource.PodRequestsAndLimits(pod)
+				ephemeralStorageLimit := podLimits[v1.ResourceEphemeralStorage]
+				sizeLimit = resource.NewQuantity(ephemeralStorageLimit.Value(), resource.BinarySI)
```

Comment:
i understand this is an upper-bound for container writable layer, logs, and empyDir.

if an emptyDir specifies a local size limit, i guess it would have had to been lower than the cumulative pod limit, and since this is tracking, but not enforcing it doesn't matter?

/cc @jingxu97

Reply:
Correct.  When we turn on enforcement we'll have to decide how to handle it.

Reply:
Yes, emptyDir has a sizeLimit too. It is used to set the upper limit of emptyDir, user should normally set it lower than the sum of pod limits. I think here you can get sizeLimit from emptyDir too if it exists, and user the smaller value.